react-validated-input-component/my-app/src/ValidateInput.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+
+export default function ValidateInput() {
+  const [inputValue, setInputValue] = useState('');
+
+  const regex = new RegExp(
+    '(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$'
+  );
+
+  let inputValueLength: number = inputValue.length;
```

Comment:
In TS you should let TS infer types as much as you can. It knows, for example, that `.length` is a number, so there's no need to specify it. It also knows that `'hidden'` is a string, so you don't need to specify that type, either.

Type inference makes your code easier to write and read, and makes it look more similar to plain JavaScript.